### PR TITLE
Modify infer_node_vals_and_errs_networkx to support nx.MultiDiGraph

### PR DESCRIPTION
**Pull request by jkausrelay**
- Update infer_node_vals_and_errs_networkx to allow a MultiDiGraph to be passed
- Can be used to infer dgs across replicates (where the replicates have the same nodes, but multiple edges with different pred_ddgs)
- Update test

### CODE REVIEW

File: timemachine/fe/mle.py [hunk 1]
```diff
@@ -224,7 +226,7 @@ def infer_node_vals_and_errs_networkx(
     node_stddev_prop: str = "inferred_dg_stddev",
     n_bootstrap: int = 100,
     seed: int = 0,
-) -> nx.DiGraph:
+) -> Union[nx.DiGraph, nx.MultiDiGraph]:
```
[the remark this answers is not in the document]
Comment (badisa) [reply timemachine/fe/mle.py:229]: I would push back against this. I find these TypeVars to be less readable than a raw Union, and I would prioritize readability over precision of typing that I don't see the clear value of.

[the remark this answers is not in the document]
Comment (jkausrelay) [reply timemachine/fe/mle.py]: Added `NxDiGraph = Union[nx.DiGraph, nx.MultiDiGraph]` as a type. There is a bit of nuance lost with this, but I think that it's more readable.

File: tests/test_mle.py [hunk 2]
```diff
@@ -299,6 +299,38 @@ def test_infer_node_vals_and_errs_networkx(nx_graph_with_reference_mle_instance)
         # assert g_res.nodes[n][node_stddev_prop] == pytest.approx(ref_dg_err)
 
 
+def test_infer_node_vals_and_errs_networkx_multi(nx_graph_with_reference_mle_instance):
```
Comment (maxentile) [inline tests/test_mle.py:302]: Not necessary, but another testing strategy could be to summarize all parallel edges as a single edge. (Since product of Gaussians is a Gaussian https://github.com/proteneer/timemachine/pull/399#issuecomment-824893476 )

File: timemachine/fe/mle.py [hunk 3]
```diff
@@ -279,7 +283,7 @@ def infer_node_vals_and_errs_networkx(
     edge_idxs = np.array(sg_relabeled.edges)
 
     dgs, dg_errs = infer_node_vals_and_errs(
-        edge_idxs,
+        np.array([e[:2] for e in edge_idxs]),
```
Comment (maxentile) [inline timemachine/fe/mle.py:286]: I found `e[:2]` surprising. Is `len(e) != 2`?

File: timemachine/fe/mle.py [hunk 4]
```diff
@@ -224,7 +228,7 @@ def infer_node_vals_and_errs_networkx(
     node_stddev_prop: str = "inferred_dg_stddev",
     n_bootstrap: int = 100,
     seed: int = 0,
-) -> nx.DiGraph:
+) -> NxDiGraph:
     """Version of :py:func:`timemachine.fe.mle.infer_node_vals_and_errs` that accepts a directed networkx graph.
```
Comment (maxentile) [inline timemachine/fe/mle.py:233]: nit: docstring falling further out of sync

**Review by maxentile — APPROVED** [review]
Looks good!

File: timemachine/fe/mle.py (hunk 3 above)
Comment (jkausrelay) [reply timemachine/fe/mle.py:286]: Will add a comment. With MultiDiGraph it's (from_node, to_node, edge_key) where edge_key = 0, 1, ... for each duplicate edge.

File: timemachine/fe/mle.py (hunk 4 above)
Comment (jkausrelay) [reply timemachine/fe/mle.py:233]: Updated

File: timemachine/fe/mle.py (hunk 3 above)
Comment (maxentile) [reply timemachine/fe/mle.py]: Ahh, then might rename this from `edge_idxs` to `edges` (since depending on input type it may or may not just contain idxs)

Comment (jkausrelay) [reply timemachine/fe/mle.py]: Updated in [65e7c57](https://github.com/proteneer/timemachine/pull/1064/commits/65e7c5797f2794ddae05eb632ed29678221ffc49)

File: timemachine/fe/mle.py [hunk 5]
```diff
@@ -276,12 +281,12 @@ def infer_node_vals_and_errs_networkx(
             ref_node_vals.append(d[ref_node_val_prop])
             ref_node_stddevs.append(d.get(ref_node_stddev_prop, 0.0))
 
-    edge_idxs = np.array(sg_relabeled.edges)
-
+    edges = np.array(sg_relabeled.edges)
+    edge_idxs = np.array([e[:2] for e in edges])  # remove edge key in MultiDiGraph if present
```
Comment (mcwitt) [inline timemachine/fe/mle.py:285]: nit: avoid array -> list -> array
```suggestion
    edge_idxs = edges[:, :2]  # remove edge key in MultiDiGraph if present
```
